cameo/strain_design/heuristic/evolutionary/optimization.py

```diff
@@ -573,24 +577,37 @@ class ReactionKnockoutOptimization(KnockoutOptimization):
     >>> knockout_optimization = ReactionKnockoutOptimization(model=model, objective_function=bpcy,
     >>>                                                      essential_reactions=["ATPM"])
     >>> knockout_optimization.run(max_evaluations=50000)
-
-
     """
 
-    def __init__(self, reactions=None, essential_reactions=None, *args, **kwargs):
+    def __init__(self, reactions=None, essential_reactions=None, use_nullspace_simplification=True, *args, **kwargs):
         super(ReactionKnockoutOptimization, self).__init__(*args, **kwargs)
         if reactions is None:
             self.reactions = set([r.id for r in self.model.reactions])
         else:
             self.reactions = reactions
         logger.debug("Computing essential reactions...")
         if essential_reactions is None:
-            self.essential_reactions = set([r.id for r in self.model.essential_reactions()])
+            self.essential_reactions = set(r.id for r in self.model.essential_reactions())
         else:
             self.essential_reactions = set([r.id for r in self.model.essential_reactions()] + essential_reactions)
 
-        exchange_reactions = set([r.id for r in self.model.exchanges])
-        self.representation = list(self.reactions.difference(self.essential_reactions).difference(exchange_reactions))
+        if use_nullspace_simplification:
+            ns = nullspace(self.model.S)
+            dead_ends = find_blocked_reactions_nullspace(self.model, ns=ns)
+            exchanges = self.model.exchanges
+            reactions = [r for r in self.model.reactions if r not in exchanges and r not in dead_ends and
+                         r.id not in self.essential_reactions]
+
+            groups = find_coupled_reactions_nullspace(self.model, ns=ns)
+            groups = [group for group in groups if any(r.id in reactions for r in group)]
+            reduced_set = reduce_reaction_set(reactions, groups)
+            to_keep = [r.id for r in reduced_set]
```

Comment:
Can't see from the code shown here if it matters down the line, but maybe make `to_keep` a set, too.